kube-hunter.py

```diff
@@ -104,8 +112,10 @@ def main():
 
         if not any(scan_options):
             if not interactive_set_config(): return
-        
+        tlock = threading.Lock()
```

Comment:
I'd prefer we do the lock changes under a separate PR, but since I'm seeing this code here I'll write some thoughts here :-) 

Note that you need just one lock to protect a variable - grab the same lock whenever you're accessing that variable (it needs to be *the exact same lock* wherever you acquire it from). So for a lock that protects a global variable, it also needs to be in global scope.

To make it absolutely obvious what the lock is for, I would 
* create the lock right next to the global variable 
* give it a more meaningful name (like lock_hunt_started or hunt_started_lock)

Reply:
Alright, I would change it :-) 
But please note I have done the locking code on a [different branch](https://github.com/aquasecurity/kube-hunter/tree/solve-multi-threading-bug) and merged it with that one so I can check it on that hunter (since the threading bug appears here).